client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -56,9 +60,15 @@ object Retry {
       }
 
     def nextAttempt(req: Request[F], attempts: Int, duration: FiniteDuration)
-                   (implicit F: Async[F], executionContext: ExecutionContext): F[DisposableResponse[F]] =
-      // TODO honor Retry-After header
-      scheduler.sleep_[F](duration).run >> prepareLoop(req.withEmptyBody, attempts + 1)
+                   (implicit F: Async[F], executionContext: ExecutionContext): F[DisposableResponse[F]] = {
+      val sleepDuration = req.headers.get(`Retry-After`).map { h =>
+        h.retry match {
+          case Left(d) => Instant.now().until(d.toInstant, ChronoUnit.SECONDS).seconds
+          case Right(secs) => secs.seconds
+        }
```

Comment:
I think I would take the max of this value and the passed in `duration` here.  I think `Retry-After` means more "wait a minimum of this time" rather than "come back at exactly this time."

Reply:
makes sense